srtcore/core.cpp

```diff
@@ -9525,65 +9525,75 @@ bool srt::CUDT::packUniqueData(CPacket& w_packet, time_point& w_origintime)
-        // after connection. No packets in the buffer, no packets are sent,
-        // no ACK to be awaited. We can screw up all the variables that are
-        // initialized from ISN just after connection.
-        //
-        // Additionally send the drop request to the peer so that it
-        // won't stupidly request the packets to be retransmitted.
-        // Don't do it if the difference isn't positive or exceeds the threshold.
-        if (packetspan > 0)
-        {
-            int32_t seqpair[2];
-            seqpair[0] = m_iSndCurrSeqNo;
-            seqpair[1] = w_packet.m_iSeqNo;
-            HLOGC(qslog.Debug, log << "... sending INITIAL DROP (ISN FIX): "
-                    << "msg=" << MSGNO_SEQ::unwrap(w_packet.m_iMsgNo) << " SEQ:"
-                    << seqpair[0] << " - " << seqpair[1] << "(" << packetspan << " packets)");
-            sendCtrl(UMSG_DROPREQ, &w_packet.m_iMsgNo, seqpair, sizeof(seqpair));
+    // Fortunately the group itself isn't being accessed.
+    if (m_parent->m_GroupOf)
+    {
+        if (m_iSndCurrSeqNo != w_packet.m_iSeqNo)
+        {
+            const int packetspan = CSeqNo::seqcmp(w_packet.m_iSeqNo, m_iSndCurrSeqNo);
+            // After increasing by 1, but being previously set as ISN-1, this should be == ISN,
+            // if this is the very first packet to send.
+            if (m_iSndCurrSeqNo == m_iISN)
+            {
+                // This is the very first packet to be sent; so there's nothing in
+                // the sending buffer yet, and therefore we are in a situation as just
+                // after connection. No packets in the buffer, no packets are sent,
+                // no ACK to be awaited. We can screw up all the variables that are
+                // initialized from ISN just after connection.
+                LOGC(qslog.Warn,
+                     log << CONID() << "packData: Fixing EXTRACTION sequence " << m_iSndCurrSeqNo
+                         << " from SCHEDULING sequence " << w_packet.m_iSeqNo << " for the first packet: DIFF="
+                         << packetspan << " STAMP=" << BufferStamp(w_packet.m_pcData, w_packet.getLength()));
+
+                // Additionally send the drop request to the peer so that it
+                // won't stupidly request the packets to be retransmitted.
+                // Don't do it if the difference isn't positive or exceeds the threshold.
+                if (packetspan > 0)
+                {
+                    int32_t seqpair[2];
+                    seqpair[0] = m_iSndCurrSeqNo;
+                    seqpair[1] = CSeqNo::decseq(w_packet.m_iSeqNo);
+                    const int32_t no_msgno = 0;
+                    LOGC(qslog.Debug,
+                         log << "packData: Sending DROPREQ (ISN FIX): SEQ: " << seqpair[0] << " - " << seqpair[1]
+                             << " (" << packetspan << " packets)");
+                    sendCtrl(UMSG_DROPREQ, &no_msgno, seqpair, sizeof(seqpair));
```

Comment:
Thank you for this important bug fix here! ⚠️ 
Sending a drop request with a non-zero message number (as it was before this fix) was a bug, because all the packets in the range of sequence numbers are intended to be dropped, not only one message.